src/java.base/share/classes/java/lang/String.java

```diff
@@ -837,13 +852,11 @@ private String(Charset charset, byte[] bytes, int offset, int length) {
                 char[] ca = new char[en];
                 int clen = ad.decode(bytes, offset, length, ca);
                 if (COMPACT_STRINGS) {
-                    byte[] bs = StringUTF16.compress(ca, 0, clen);
-                    if (bs != null) {
-                        value = bs;
-                        coder = LATIN1;
-                        return;
-                    }
+                    byte[] val = StringUTF16.compress(ca, 0, clen);;
+                    this.coder = StringUTF16.coderFromArrayLen(val, clen);
+                    this.value = val;
                     initCompressionFlag();
```

Comment:
This should be conditional (it should have been before).

Reply:
Fixed